### DISCUSSION TITLE
docker-machine: switch to GitLab fork

### DISCUSSION THREAD

**Discussion by dolmen** [Homebrew/discussions#2213] [Writing Formulae/Casks]
The upstream project `docker-machine` is [in maintenance mode](https://github.com/docker/machine/issues/4537).

However GitLab maintains [a fork where bugs are fixed](https://docs.gitlab.com/runner/executors/docker_machine.html#forked-version-of-docker-machine):
* Make DigitalOcean driver RateLimit aware
* Add backoff to Google driver operations check
* Add --google-min-cpu-platform option for machine creation
* Use cached IP for Google driver
* Use cached IP for AWS driver
* Add support for using GPUs in Google Compute Engine
* Support running AWS instances with IMDSv2 

It would be helpful for homebrew users to have easy access to those fixes.

So, what about switching the [docker-machine formulae](https://github.com/Homebrew/homebrew-core/blob/HEAD/Formula/docker-machine.rb) to the GitLab fork?

Repository: https://gitlab.com/gitlab-org/ci-cd/docker-machine
Binaries: https://gitlab-docker-machine-downloads.s3.amazonaws.com/main/index.html

**Comment by SMillerDev** [comment]
If the original upstream considers this their successor we can switch.

**Comment by dolmen** [comment]
> If the original upstream considers this their successor we can switch.

OK, so this will never happen.

What is the step forward to get that GitLab fork available on HomeBrew?

Should I propose a `docker-machine-gitlab` recipe that will declare an install conflict with the `docker-machine` recipe? Is there a naming convention for publishing a recipe of an alternative implementation of a popular project?

**Reply by SMillerDev** [reply]
Additionally if there is consencus in the package management world that this is the successor we can also switch. Something like Fedora or Debian switching would indicate that.

As for the alternate Formula, usually projects rename their fork so I don't think there is enough of a precedent to have a naming convention.
